lib/init.c

```diff
@@ -296,31 +296,41 @@ void rpc_set_auxiliary_gids(struct rpc_context *rpc, uint32_t len, uint32_t* gid
 void rpc_set_error(struct rpc_context *rpc, const char *error_string, ...)
 {
         va_list ap;
-	char *old_error_string;
```

Comment:
I don't think we can do this change and drop old_error_string.
The reason is it is useful to be able to append the previous error strong to the new one being generated.

For example instances like this:
                smb2_set_error(smb2, "Failed to create set command. %s",
                               smb2_get_error(smb2));

So undo this change.

Reply:
good point! will update